policyengine_core/simulations/simulation.py

```diff
@@ -1396,77 +1414,31 @@ def extract_person(
 
         return json.loads(json.dumps(situation, cls=NpEncoder))
 
-    def _get_macro_cache(
-        self,
-        variable_name: str,
-        period: str,
-    ):
+    def check_macro_cache(self, variable_name: str, period: str) -> bool:
         """
-        Get the cache location of a variable for a given period, if it exists.
+        Check if the variable is able to have cached value
         """
-        if not self.is_over_dataset:
-            return None
+        if hasattr(self, "dataset"):
```

Comment:
```suggestion
        if hasattr(self, "dataset") and self.dataset.data_format == Dataset.FLAT_FILE:
```

I think this is better form instead of nesting if's